awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,178 @@
+                models.Schedule, models.CustomInventoryScript,
+                models.NotificationTemplate):
+        counts[camelcase_to_underscore(cls.__name__)] = cls.objects.count()
+
+    venvs = get_custom_venv_choices()
+    counts['custom_virtualenvs'] = len([
+        v for v in venvs
+        if os.path.basename(v.rstrip('/')) != 'ansible'
+    ])
+
+    inv_counts = dict(models.Inventory.objects.order_by().values_list('kind').annotate(Count('kind')))
+    inv_counts['normal'] = inv_counts[''] # rename '' to 'normal'
+    counts['inventories'] = inv_counts
+    
+    counts['active_host_count'] = models.Host.objects.active_count()   
+    active_sessions = Session.objects.filter(expire_date__gte=now()).count()
+    api_sessions = models.UserSessionMembership.objects.select_related('session').filter(session__expire_date__gte=now()).count()
+    channels_sessions = active_sessions - api_sessions
+    counts['active_sessions'] = active_sessions
+    counts['active_api_sessions'] = api_sessions
+    counts['active_channels_sessions'] = channels_sessions
+    counts['running_jobs'] = models.UnifiedJob.objects.filter(status__in=('running', 'waiting',)).count()
+    return counts
+
+    
+@register('org_counts')
+def org_counts(since):
+    counts = {}
+    for org in models.Organization.objects.annotate(num_users=Count('member_role__members', distinct=True), 
+                                                    num_teams=Count('teams', distinct=True)).values('name', 'id', 'num_users', 'num_teams'):
+        counts[org['id']] = {'name': org['name'],
+                            'users': org['num_users'],
+                            'teams': org['num_teams']
+                            }
+    return counts
+    
+    
+@register('cred_type_counts')
+def cred_type_counts(since):
+    counts = {}
+    for cred_type in models.CredentialType.objects.annotate(num_credentials=Count('credentials', distinct=True)):  
+        counts[cred_type.id] = {'name': cred_type.name,
+                                'credential_count': cred_type.num_credentials
+                                }
+    return counts
+    
+    
+@register('inventory_counts')
+def inventory_counts(since):
+    counts = {}
+    for inv in models.Inventory.objects.filter(kind='').annotate(num_sources=Count('inventory_sources', distinct=True), 
+                                                 num_hosts=Count('hosts', distinct=True)).only('id', 'name', 'kind'):
+        counts[inv.id] = {'name': inv.name,
+                          'kind': inv.kind,
+                          'hosts': inv.num_hosts,
+                          'sources': inv.num_sources
+                          }
+
+    for smart_inv in models.Inventory.objects.filter(kind='smart'):
+        counts[smart_inv.id] = {'name': smart_inv.name,
+                          'kind': smart_inv.kind,
+                          'num_hosts': smart_inv.hosts.count(),
```

Comment:
Don't forget to give these the same treatment as the annotation above.  In fact, it looks like it could be done in a single loop.

Reply:
thanks for catching that. I'll add that to my PR.

Reply:
Oh, I think I see what you mean about smart inventories and hosts that you mentioned in one of the other comments.  I'd push for making a change to that through-table so that you can do the counts as an annotation.

Reply:
@chrismeyersfsu can you weigh in on this performance-wise?

Reply:
@wwitzel3 took a look at this with me and did query analysis on this as is compared to it as one consolidated query and it performed better as is.